pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -424,6 +424,13 @@ func translateRoute(
 		applyDirectResponse(out, in.DirectResponse)
 	} else {
 		applyHTTPRouteDestination(out, node, virtualService, in, mesh, authority, serviceRegistry, listenPort, hashByDestination)
+		if node.Type == model.Router {
```

Comment:
Users can already set the name as part of the virtual service API. So this is going to overwrite that which is not desired; explicit API must take precedence over implicit.

IMO we should not do this at all though - if users do not want `default` they should use the explicit naming. Otherwise we have a very poor heuristic which may not even work all the time (GetCluster is often `""`)

Reply:
I'm wrong, [DefaultRouteName ](https://github.com/istio/istio/blob/8a3b838f788965dd1027d38e7b55664abf8989dc/pilot/pkg/networking/core/v1alpha3/route/route.go#L63) will be used when no vs.